agent/jvm/src/test/java/org/jolokia/jvmagent/client/util/VirtualMachineHandlerTest.java

```diff
@@ -56,6 +56,11 @@ public void emptyPid() {
 
     @Test
     public void listAndAttach() throws Exception, NoSuchMethodException, IllegalAccessException {
+        // FIXME Fails on the IBM JVM; however, the agent seems to work. Skip for now just to get the tests to run on the IBM JVM. - Anyone?
```

Comment:
Any idea why and where this fails ?

Reply:
Where it fails? Directly at the first assert (when it checks if the process-collection is not empty). Why it fails I am honestly not sure about (as I myself very very rarely use J9). However, my theory here is that we are running into the same "lack of attach-mode"-problem as with JMockit. I know the class exists otherwise it would fail earlier and when I tried to find a reason for this I did a fair bit of googling and it seems that the general consensus in that respect is that you simply can't attach to J9 the same way you can with OpenJDK/Oracle JDK. As I wrote in the comment, Jolokia itself seems to work just fine after a fair amount of manual testing but it's naturally a bit disturbing to have to disable a test. I could look into it once more during the weekend, perhaps I'll find the root-cause.

Reply:
Not much for my theory it seemed. "attach" worked fine, "detach" failed with IllegalAccessException, sent another commit your way. Think all your requested changes are resolved now.